### PR TITLE
fix: loading of session form in wizard step 5

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes https://github.com/fossasia/open-event-frontend/issues/5768

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/routes/events/view/edit/sessions-speakers.js [hunk 1]
```diff
@@ -14,7 +14,19 @@ export default class SessionsSpeakersRoute extends Route.extend(EventWizardMixin
     const tracksPromise = data.event.get('tracks');
     const microlocationsPromise = data.event.get('microlocations');
     const sessionTypesPromise = data.event.get('sessionTypes');
-    const speakersCallPromise = this.getOrCreate(data.event, 'speakersCall', 'speakers-call');
+    const speakersCallPromise = new RSVP.Promise(resolve => {
+      data.event
+        .get('speakersCall')
+        .then(relationshipRecord => {
+          resolve(relationshipRecord);
```
Comment (iamareebjamal) [inline app/routes/events/view/edit/sessions-speakers.js:21]: This then call is redundant

File: app/routes/events/view/edit/sessions-speakers.js [hunk 2]
```diff
@@ -14,7 +14,19 @@ export default class SessionsSpeakersRoute extends Route.extend(EventWizardMixin
     const tracksPromise = data.event.get('tracks');
     const microlocationsPromise = data.event.get('microlocations');
     const sessionTypesPromise = data.event.get('sessionTypes');
-    const speakersCallPromise = this.getOrCreate(data.event, 'speakersCall', 'speakers-call');
+    const speakersCallPromise = new RSVP.Promise(resolve => {
```
Comment (iamareebjamal) [inline app/routes/events/view/edit/sessions-speakers.js:17]: Can you please move this to an async function, and use await in that function?

File: app/routes/events/view/edit/sessions-speakers.js (hunk 1 above)
Comment (maze-runnar) [reply app/routes/events/view/edit/sessions-speakers.js:21]: after removing then, the session page isn't loading

File: app/routes/events/view/edit/sessions-speakers.js [hunk 3]
```diff
@@ -14,7 +14,19 @@ export default class SessionsSpeakersRoute extends Route.extend(EventWizardMixin
     const tracksPromise = data.event.get('tracks');
     const microlocationsPromise = data.event.get('microlocations');
     const sessionTypesPromise = data.event.get('sessionTypes');
-    const speakersCallPromise = this.getOrCreate(data.event, 'speakersCall', 'speakers-call');
+    const speakersCallPromise = new RSVP.Promise(async resolve => {
+      await data.event
```
Comment (iamareebjamal) [inline app/routes/events/view/edit/sessions-speakers.js:18]: Don't await here, it's useless and will block the loading. Move these statements in another function and await in that function. No await should be there in model function

File: app/routes/events/view/edit/sessions-speakers.js [hunk 4]
```diff
@@ -58,4 +46,21 @@ export default class SessionsSpeakersRoute extends Route.extend(EventWizardMixin
...
+          resolve(relationshipRecord);
+        })
+        .catch(() => {
+          const record = this.store.createRecord('speakers-call', {
+            event
+          });
+          resolve(record);
+        });
+    });
+    return cfs;
```
Comment (iamareebjamal) [inline app/routes/events/view/edit/sessions-speakers.js:64]:
```suggestion
	try {
	  return await event.get('speakersCall');
	} catch {
	  return this.store.createRecord('speakers-call', {
        event
      });
	}
```

File: app/routes/events/view/edit/sessions-speakers.js [hunk 5]
```diff
@@ -14,7 +14,7 @@ export default class SessionsSpeakersRoute extends Route.extend(EventWizardMixin
     const tracksPromise = data.event.get('tracks');
     const microlocationsPromise = data.event.get('microlocations');
     const sessionTypesPromise = data.event.get('sessionTypes');
-    const speakersCallPromise = this.getOrCreate(data.event, 'speakersCall', 'speakers-call');
+    const speakersCallPromise =  this.getOrCreateCFS(data.event);
```
Comment (iamareebjamal) [inline app/routes/events/view/edit/sessions-speakers.js:17]:
```suggestion
    const speakersCallPromise = this.getOrCreateCFS(data.event);
```

**Comment by mariobehling** [comment]
Working for me, but the issue did not appear all the time, just from time to time. We need to test this a while. So, I think this can be merged then.

**Pull request merged**